### PR TITLE
Reduce CMake spam

### PR DESCRIPTION
**Pull request by klimkin**
Changelog: Feature: Use CMAKE_MESSAGE_LOG_LEVEL to control the amount of output of from cmake files

Note, `CMAKE_MESSAGE_LOG_LEVEL` requires CMake 3.17. For older versions the messages will be dropped. It's possible to have  make a workaround in `conan_message()` to convert the type to `STATUS` for any older version.
https://cmake.org/cmake/help/latest/variable/CMAKE_MESSAGE_LOG_LEVEL.html

Solves https://github.com/conan-io/conan/issues/8145

From this:
~~~
-- Conan: Using autogenerated FindSystemCLanguage.cmake
-- Found SystemCLanguage: 2.3.3 (found version "2.3.3")
-- Library systemc found .conan/data/systemc/2.3.3/syssim/stable/package/ecec4cd1dc0d521a95bc6a23f7972092d3c15d7f/lib/libsystemc.so
-- Found: .conan/data/systemc/2.3.3/syssim/stable/package/ecec4cd1dc0d521a95bc6a23f7972092d3c15d7f/lib/libsystemc.so
-- Library systemc found .conan/data/systemc/2.3.3/syssim/stable/package/ecec4cd1dc0d521a95bc6a23f7972092d3c15d7f/lib/libsystemc.so
-- Found: .conan/data/systemc/2.3.3/syssim/stable/package/ecec4cd1dc0d521a95bc6a23f7972092d3c15d7f/lib/libsystemc.so
-- Conan: Using autogenerated FindGTest.cmake
-- Found GTest: 1.10.0 (found version "1.10.0")
-- Library gtest_main found .conan/data/gtest/1.10.0/_/_/package/00673baad6d72163dd851981a9d73649926a72ff/lib/libgtest_main.a
-- Found: .conan/data/gtest/1.10.0/_/_/package/00673baad6d72163dd851981a9d73649926a72ff/lib/libgtest_main.a
-- Library gmock_main found .conan/data/gtest/1.10.0/_/_/package/00673baad6d72163dd851981a9d73649926a72ff/lib/libgmock_main.a
-- Found: .conan/data/gtest/1.10.0/_/_/package/00673baad6d72163dd851981a9d73649926a72ff/lib/libgmock_main.a
...
~~~

To this:
~~~
-- Found SystemCLanguage: 2.3.3 (found version "2.3.3")
-- Found GTest: 1.10.0 (found version "1.10.0")
~~~

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by solvingj** [comment]
It's definitely a nice thing to control.  It should probably be configurable via `[conf]`.

**Review by memsharded — COMMENTED** [review]
Conan 2.0 will provide support for CMake 3.15, and current Conan 1.X supports in some cases down to CMake 2.8, and in the find_package generator, something like 3.3 I think is the supported case.

Changes done need to take this into account, please don't change if it will not work properly in older CMake versions. Everything in ``conan.tools.xxxx`` can be considered 2.0-oriented, and thus can contain CMake 3.15 only features, but other generators like ``cmake_find_package`` must guarantee backwards compatibility with older CMake versions.

**Comment by solvingj** [comment]
@memsharded comments on using `[conf]`?

**Review by memsharded — CHANGES_REQUESTED** [review]
Hi @klimkin 

If you would like to move this forward:

- This cannot rely on https://cmake.org/cmake/help/latest/variable/CMAKE_MESSAGE_LOG_LEVEL.html, because that is from cmake 3.17.
- New integrations in ``conan.tools.cmake`` will need to support CMake 3.15
- Do not touch the previous integrations, as they need to support even older CMake versions, and will be removed in Conan 2.0, not worth the risk.

I understand that probably not worth to do a custom solution for this, or too much effort, please tell.

**Comment by memsharded** [comment]
Ok, as this is a bit risky (depends on CMake>3.17, which is breaking for many users), cannot be merged. So I propose the following plan:

- Lets close this PR, it is a bit outdated and conflicted.
- Conan 2.0 is now refactoring the api and the output. The output will include logging and verbosity levels.
- Try to align the default verbosity levels of build systems with the Conan new levels
- Possibly propose a possibility of setting build-system logging levels (general ones, not only CMake), with different values than the Conan logging levels
- Do this only for the new generators, better not touch the legacy ones, not worth the risk

In short, we like the feature and we will implement i , (without modifying the legacy ones, only for the new). Thanks very much for your contribution

**Comment by klimkin** [comment]
That's fine, thank you for reviewing this. Currently, there is a middle-ground solution to suppress spam using `set(CONAN_CMAKE_SILENT_OUTPUT TRUE)`. So you can keep focus to get better logging in 2.0.
